Kernel/Devices/HID/USB/MouseDevice.cpp

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2022, Liav A. <liavalb@hotmail.co.il>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/Try.h>
+#include <AK/TypedTransfer.h>
+#include <Kernel/Bus/USB/USBDescriptors.h>
+#include <Kernel/Bus/USB/USBRequest.h>
+#include <Kernel/Bus/USB/USBTransfer.h>
+#include <Kernel/Devices/DeviceManagement.h>
+#include <Kernel/Devices/HID/USB/MouseDevice.h>
+#include <Kernel/Library/KString.h>
+
+namespace Kernel {
+
+ErrorOr<NonnullRefPtr<USBMouseDevice>> USBMouseDevice::try_create_instance(USB::Device const& usb_device)
+{
+    auto device = TRY(DeviceManagement::try_create_device<USBMouseDevice>(usb_device));
+    TRY(device->create_and_start_polling_process());
+    return *device;
+}
+
+ErrorOr<void> USBMouseDevice::create_and_start_polling_process()
+{
+    VERIFY(m_attached_usb_device->interrupt_pipe());
+    [[maybe_unused]] auto interrupt_in_transfer = TRY(static_cast<USB::InterruptInPipe*>(m_attached_usb_device->interrupt_pipe())->submit_interrupt_in_transfer(6, 10, [this](auto* transfer) {
+        if (transfer->transfer_data_size() == 0)
```

Comment:
Should we check for `transfer->complete()` instead? Or we could also argue that it doesn't make sense to call the callback when we receive NAKs.

Reply:
We probably shouldn't send transfers if we got a NAK in my opinion. @Quaker762 what do you think about it?

Reply:
I think that if a transfer ends with 0 bytes being transferred, it's probably not meaningful to invoke the async callback, hence I added a commit to just discard such transfer.